rs/tests/nested/src/lib.rs

```diff
@@ -152,10 +152,13 @@ pub fn upgrade_hostos(env: TestEnv) {
         vec![node_id],
     ));
 
+    info!(logger, "Orchestrator dashboard initial health check...");
+    host.await_orchestrator_dashboard_accessible().unwrap();
+
     // The HostOS upgrade is applied with a reboot to the host machine.
-    // Wait for the host to reboot before checking Orchestrator dashboard status
+    // Wait for the host to reboot causing the Orchestrator dashboard to be inaccessible.
```

Comment:
What about using an explicit check for host reboot by retrieving the current boot ID before calling `update_nodes_hostos_version` using:

```
sudo journalctl --list-boots | tail -n1 | cut -d' ' -f4
```

which return something like `d895280ed73c4d608e139418759e075e`. 

And then waiting for that ID to change after the call to `update_nodes_hostos_version`?

Reply:
Implemented in: https://github.com/dfinity/ic/pull/4724/files